File: admin/frontend/src/types/comment.ts
import type { BaseResponse } from './common'

// 评论状态枚举
export enum CommentStatus {
  All = 'all',
  Pending = 'pending',
  Approved = 'approved',
  Spam = 'spam'
}

// 评论状态选项
export const COMMENT_STATUS_OPTIONS = [
  { label: '全部', value: CommentStatus.All },
  { label: '待审核', value: CommentStatus.Pending },
  { label: '已通过', value: CommentStatus.Approved },
  { label: '垃圾评论', value: CommentStatus.Spam }
]

// 评论查询参数
export interface CommentQuery {
  keyword?: string
  article_title?: string
  status?: CommentStatus
  start_date?: string
  end_date?: string
  include_replies?: boolean
  only_root?: boolean
  page?: number
  page_size?: number
  sort_field?: string
  sort_order?: 'ascend' | 'descend'
}

// 评论响应数据
export interface CommentResponse {
  id: number
  content: string
  article_id: number
  article_title: string
  user_id: number
  user_name: string
  parent_id: number | null
  is_approved: boolean
  is_spam: boolean
  reply_count: number
  like_count: number
  ip_address?: string
  user_agent?: string
  created_at: string
  updated_at: string
  replies?: CommentResponse[]
  level?: number
}

// 评论创建参数
export interface CommentCreate {
  content: string
  article_id: number
  parent_id?: number
}

// 评论更新参数
export interface CommentUpdate {
  content?: string
  is_approved?: boolean
  is_spam?: boolean
  parent_id?: number | null
}

// 评论树节点
export interface CommentTreeNode extends CommentResponse {
  children?: CommentTreeNode[]
  level: number
}

// 评论批量操作参数
export interface CommentBatchActionParams {
  comment_ids: number[]
}

// 评论统计信息
export interface CommentStats {
  total: number
  pending: number
  approved: number
  spam: number
  replies: number
}

// API 响应类型
export type CommentApiResponse = BaseResponse<CommentResponse>
export type CommentListResponse = BaseResponse<{
  items: CommentResponse[]
  total: number
}>
export type CommentTreeResponse = BaseResponse<CommentTreeNode[]>
export type CommentStatsResponse = BaseResponse<CommentStats>

File: admin/backend/app/schemas/comment.py
from pydantic import BaseModel, ConfigDict
from typing import Optional, List
from datetime import datetime

class CommentBase(BaseModel):
    content: str
    article_id: int
    parent_id: Optional[int] = None
    is_approved: Optional[bool] = False
    is_spam: Optional[bool] = False

class CommentCreate(CommentBase):
    pass

class CommentUpdate(CommentBase):
    content: str

class CommentQuery(BaseModel):
    keyword: Optional[str] = None
    status: Optional[str] = None
    article_id: Optional[int] = None
    article_title: Optional[str] = None
    user_id: Optional[int] = None
    start_date: Optional[datetime] = None
    end_date: Optional[datetime] = None
    include_replies: Optional[bool] = True  # 是否包含回复
    only_root: Optional[bool] = False  # 是否只查询根评论
    page: int = 1
    size: int = 10

class CommentResponse(CommentBase):
    id: int
    article_id: int
    user_id: int
    ip_address: Optional[str] = None
    user_agent: Optional[str] = None
    created_at: datetime
    updated_at: Optional[datetime] = None
    is_approved: bool = False
    is_spam: bool = False
    like_count: int = 0
    user_name: Optional[str] = None
    article_title: Optional[str] = None
    reply_count: int = 0  # 回复数量
    replies: Optional[List['CommentResponse']] = None  # 回复列表

    model_config = ConfigDict(from_attributes=True)

# 解决循环引用问题
CommentResponse.model_rebuild() 